test/worker_test.py

```diff
@@ -1296,3 +1298,358 @@ def store_task(t, error_msg):
         w._handle_next_task()
 
         self.assertEqual(result, [task])
+
+
+class PerTaskRetryPolicyBehaviorTest(unittest.TestCase):
+    def setUp(self):
+        self.per_task_retry_count = 2
+        self.default_retry_count = 1
+        self.sch = Scheduler(retry_delay=0.1, retry_count=self.default_retry_count, prune_on_get_work=True, record_task_history=False)
+
+        self.test_instance_id = str(uuid4())
+
+    def test_with_all_disabled_with_single_worker(self):
+        """
+            With this test, a case which has a task (A), requires two another tasks (B,C) which both is failed, is tested.
+
+            Task C has default retry_count which is 1, but Task B has retry_count at task level as 2.
+
+            This test is running on single worker
+        """
+        class C(luigi.Task):
+            task_namespace = self.test_instance_id
+
+            retry_index = 0
+
+            def run(self):
+                self.retry_index += 1
+                raise Exception("Retry index is %s for %s" % (self.retry_index, self.task_family))
+
+        c = C()
+
+        class B(Task):
+            task_namespace = self.test_instance_id
+
+            retry_count = self.per_task_retry_count
+
+            retry_index = 0
+
+            def run(self):
+                self.retry_index += 1
+                raise Exception("Retry index is %s for %s" % (self.retry_index, self.task_family))
+
+        b = B()
+
+        class A(luigi.WrapperTask):
+            task_namespace = self.test_instance_id
+
+            def requires(self):
+                return [b, c]
+
+        a = A()
+
+        with Worker(scheduler=self.sch, worker_id='X', keep_alive=True, wait_interval=0.1) as w1:
+            self.assertTrue(w1.add(a))
+
+            self.assertFalse(w1.run())
+
+            self.assertEqual([a.task_id], list(self.sch.task_list('PENDING', 'UPSTREAM_DISABLED').keys()))
+
+            self.assertEqual(sorted([b.task_id, c.task_id]), sorted(self.sch.task_list('DISABLED', '').keys()))
```

Comment:
Sometimes, ordering may change. For example, I run tests in PyCharm and also with tox. Ordering the elements in PyCharm may be different the ordering in tox. This is why I use `sorted`.